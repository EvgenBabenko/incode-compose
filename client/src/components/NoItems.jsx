import React from 'react';
import T from 'prop-types';

import { withStyles } from '@material-ui/core/styles';

const styles = {
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
};

const NoItems = ({ text, children, classes }) => (
  <div className={classes.root}>
    <p>
      {text}
    </p>
    {children}
  </div>
);

NoItems.propTypes = {
  classes: T.objectOf(T.string).isRequired,
  text: T.string,
  children: T.element,
};

NoItems.defaultProps = {
  text: '',
  children: null,
};

export default withStyles(styles)(NoItems);
