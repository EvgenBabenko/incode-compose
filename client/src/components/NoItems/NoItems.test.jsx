import React from 'react';
import { render, mount } from 'enzyme';

import NoItems from './NoItems';

describe('>>>NoItems', () => {
  let wrapper;
  const text = 'No items';
  const child = 'Child';

  const item = (
    <NoItems text={text}>
      <div>
        {child}
      </div>
    </NoItems>
  );

  it('render the component', () => {
    wrapper = render(item);

    expect(wrapper.length).toEqual(1);

    expect(wrapper).toMatchSnapshot();
  });

  it('formats text correct', () => {
    wrapper = render(item);

    expect(wrapper.find('p').text()).toEqual(text);
  });

  it('contains child component', () => {
    wrapper = mount(item);

    expect(wrapper.contains(
      <div>
        {child}
      </div>,
    )).toBe(true);
  });
});
