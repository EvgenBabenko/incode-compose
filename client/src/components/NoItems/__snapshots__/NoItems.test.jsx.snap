// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>NoItems render the component 1`] = `
<div
  class="NoItems-root-1"
>
  <p>
    No items
  </p>
  <div>
    Child
  </div>
</div>
`;
